### DISCUSSION TITLE
Storage last directory location

### DISCUSSION THREAD

**Discussion by Joandres-Lara** [microsoft/terminal#13453] [Q&A] [answered]
When you close the terminal and change the directory location to C:/path/to/location and reopen it, it appears in C:/windows/system32, and when you open a new terminal, it also opens in the location default C:/windows/system32, and maybe it should open in C:/path/to/location, for this it may be necessary to use a store of paths, so that they are shared between the different similar windows.

**Comment by WSLUser** [comment]
So you can actually set the common working directory with OSC 9; 9, a ConEmu sequence. Not quite the same but it would allow you to change the directory upon re-launching WT. It sounds like you want session management similar to `screen`. Not sure if there are any TWG  specs out there for that. I think your best bet is to use a session aware application like tmux or screen. If it need to be strictly Windows, I'm not familiar with what the Windows equivalent of those would be. I wonder if ConEmu, Mintty or some other Windows-based terminal has this feature.

**Reply by Joandres-Lara** [reply]
@WSLUser Thank you for answering me, I think it would be much more practical if the solution were native, I understand that there is no standard, but it could be a great help, to restart where you left your work, at least the directory.

**Reply by zadjii-msft** [reply]
https://github.com/microsoft/terminal/issues/3158#issuecomment-628628120 has some discussion as to why we can't just do this by default on Windows. Many shells don't actually change the CWD they're in, so we ultimately need to rely on the shell to tell us some other way.

**Comment by zadjii-msft** [comment, marked as answer]
Refer to:

* https://docs.microsoft.com/en-us/windows/terminal/tutorials/new-tab-same-directory
* https://docs.microsoft.com/en-us/windows/terminal/customize-settings/startup#behavior-when-starting-a-new-terminal-session

You can configure the shell to tell the Terminal about the CWD. Once you enable "Open windows from a previous session", the Terminal will automatically reopen in that directory

**Reply by Joandres-Lara** [reply]
@zadjii-msft Thanks for answers me, this is great, i think that's what i was looking for.
